### PR TITLE
tsc: fix param error.

### PR DESCRIPTION
**Pull request by peaceshi**

### CODE REVIEW

**Review by ibgreen — APPROVED** [review]

File: modules/core/src/controllers/controller.js [hunk 1]
```diff
@@ -688,8 +709,8 @@ export default class Controller {
     return true;
   }
 
-  _getTransitionProps() {
+  _getTransitionProps(TransitionProps = {}) {
     // Transitions on double-tap and key-down are only supported by MapController
-    return NO_TRANSITION_PROPS;
+    return TransitionProps ?? NO_TRANSITION_PROPS;
```
Comment (Pessimistress) [inline modules/core/src/controllers/controller.js:714]: Why is a breaking change made here?

**Review by Pessimistress — COMMENTED** [review]
The class should be typed by adding a .d.ts file. Code change to .js should be minimized to avoid unintended breakages. You can find the typing of a similar class in the react-map-gl library: https://github.com/visgl/react-map-gl/blob/master/src/utils/map-controller.d.ts

File: modules/core/src/controllers/controller.js (hunk 1 above)
Comment (peaceshi) [reply modules/core/src/controllers/controller.js:714]: `this.updateViewport(A,B,C)`

B: {}
```
this._getTransitionProps({around: pos});
this._getTransitionProps();
NO_TRANSITION_PROPS;
null (changed to {})
```

**Comment by peaceshi** [comment]
> The class should be typed by adding a .d.ts file. Code change to .js should be minimized to avoid unintended breakages. You can find the typing of a similar class in the react-map-gl library: https://github.com/visgl/react-map-gl/blob/master/src/utils/map-controller.d.ts

[With TypeScript 3.7](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#--declaration-and---allowjs),
TypeScript added support for generating .d.ts files from JavaScript using JSDoc syntax.

This set up means you can own the editor experience of TypeScript-powered editors **without** porting your project to TypeScript, or **having to maintain .d.ts files in your codebase**.
TypeScript supports most JSDoc tags, you can find [the reference here](https://www.typescriptlang.org/docs/handbook/type-checking-javascript-files.html#supported-jsdoc).

**Comment by ibgreen** [comment]
> The class should be typed by adding a .d.ts file. Code change to .js should be minimized to avoid unintended breakages. 

FWIW, my guidance has been an incremental adoption strategy as outlined in #5589 , and the goal at this point was just to get the existing JS code base (initially core module to pass typescript checks). This should mostly require making small changes to JS code. 

My intention was that reviews should focus on making sure such changes are not breaking. Adding `.d.ts` files will come later.

Also, mixing JSDoc typings and typescript has pretty been much required in other frameworks when using `d.ts` files, I do not see major issues with this, especially in the initial stage. There is ongoing work to make ocular-dev-tools to support `.ts` directly, when that happens we can revisit and potentially move completely to TS, but regardless the first step is still need to make sure we fix the basic typescript errors in the JS code.

**Pull request closed without merging**